src/lib/tls/tls_channel.h

```diff
@@ -227,14 +244,20 @@ class BOTAN_DLL Channel
 
       const Handshake_State* pending_state() const { return m_pending_state.get(); }
 
+      /* methods to handle incoming traffic through Channel::receive_data. */
+      void process_handshake_ccs(secure_vector<byte>& record,
+                                 u64bit& record_sequence,
+                                 Record_Type& record_type,
+                                 Protocol_Version& record_version);
+
+      void process_application_data(secure_vector<byte>& record);
+
+      void process_alert(secure_vector<byte>& record);
+
       bool m_is_datagram;
 
       /* callbacks */
-      data_cb m_data_cb;
-      alert_cb m_alert_cb;
-      output_fn m_output_fn;
-      handshake_cb m_handshake_cb;
-      handshake_msg_cb m_handshake_msg_cb;
+      Callbacks m_callbacks;
```

Comment:
This should be a reference (`Callbacks&`) in order for the virtual functions to work.